### PR TITLE
Add endpoint for direct submission of Ethereum events

### PR DESCRIPTION
**Pull request by james-chf**
Closes https://github.com/anoma/namada/issues/227

When the `namada_apps/eth-fullnode` feature is not enabled, there is a config option - `ledger.ethereum.oracle_event_endpoint` - which if set to true, makes it so that a server is started by the oracle exposing an endpoint at `http://127.0.0.1:3030` for submission of Borsh-serialized `EthereumEvent`s. Any events submitted there will be included in validator's vote extensions.

Enabling this functionality at runtime can be done like so:

```shell
ANOMA_LEDGER__ETHEREUM__ORACLE_EVENT_ENDPOINT=true target/debug/namadan ledger run
```

This functionality can be tested out using a devtool built from https://github.com/anoma/devtool/pull/2

```shell
devtool submit-fake-transfer-to-namada \
	--nonce 1 \
	--receiver atest1v4ehgw36x5myzdfcg5myg3feg4p5vvzyxfrrgdpnxccnvvp5gdrrqse3gdz5zv2yxsenqs3ntgy89n
```

### CODE REVIEW

File: apps/src/lib/config/ethereum.rs [hunk 1]
```diff
@@ -8,12 +8,20 @@ pub struct Config {
     /// The Ethereum JSON-RPC endpoint that the Ethereum event oracle will use
     /// to listen for events from the Ethereum bridge smart contracts
     pub oracle_rpc_endpoint: String,
+    /// If this is set to `true`, then instead of the oracle listening for
+    /// events at a Ethereum JSON-RPC endpoint, an endpoint will be exposed by
+    /// the ledger for submission of Borsh-serialized
+    /// [`namada::types::ethereum_events::EthereumEvent`]s
```
Comment (sug0) [inline apps/src/lib/config/ethereum.rs:14]: nit: you can add a `use` statement with the imported path, accompanied by a `#[allow(unused_imports)]` tag. the generated docs reference the full path, with the import this would be shorter.

**Review by sug0 — APPROVED** [review]
lgtm

File: apps/src/lib/config/ethereum.rs (hunk 1 above)
Comment (james-chf) [reply apps/src/lib/config/ethereum.rs]: It feels odd importing something just to use it in the docstring, but it makes sense cc https://github.com/rust-lang/rust/issues/79542

File: apps/src/lib/config/ethereum.rs [hunk 2]
```diff
@@ -1,3 +1,6 @@
+//! Configuration settings to do with the Ethereum bridge.
+#[allow(unused_imports)]
+use namada::types::ethereum_events::EthereumEvent;
```
Comment (batconjurer) [inline apps/src/lib/config/ethereum.rs:3]: I'm confused, why is this import here?

**Review by batconjurer — APPROVED** [review]
Small question, but otherwise looks good

File: apps/src/lib/config/ethereum.rs (hunk 2 above)
Comment (james-chf) [reply apps/src/lib/config/ethereum.rs:3]: https://github.com/anoma/namada/pull/294#discussion_r962311096

(I'm happy to use either the qualified or unqualified symbol in the docstring)

Comment (batconjurer) [reply apps/src/lib/config/ethereum.rs:3]: oh wow. Ok.
